src/Logary/Metric/Gauge.fs

```diff
@@ -0,0 +1,79 @@
+namespace Logary.Metric
+
+
+type GaugeConf =
+  {
+   basicInfo: BasicConf
+   histogramConf: HistogramConf option
+  }
+
+  interface MetricBuilder<IGauge> with
+    member x.build labels registry =
+      let histrgomMetric =
+        x.histogramConf
+        |> Option.map (fun conf ->
+          let labelValues = conf.basicInfo.labelNames |> Array.map (fun name -> labels.[name])
+          conf |> registry.registerMetric |> Metric.labels labelValues)
+      new Gauge(x, labels, histrgomMetric) :> IGauge
+
+    member x.basicConf = x.basicInfo
+
+  static member create (name, description) =
+    let basic = { name =  name; description = description; labelNames = [||] }
+    {basicInfo = basic; histogramConf = None}
+
+  static member create (basic) =
+    {basicInfo = basic; histogramConf = None}
+
+and Gauge(conf, labels, histogram) =
+
+  let mutable gaugeValue = new DoubleAdder()
+  let changeGaugeValueLock = obj ()
+
+  let tryDoHistogramObserve () =
+    histogram |> Option.iter(fun histogram ->
+      let sum = gaugeValue.Sum ()
+      histogram.observe sum)
+
+  interface IGauge with
+    member x.inc value =
```

Comment:
This is a counter, not a gauge? https://godoc.org/github.com/prometheus/client_golang/prometheus#hdr-A_Basic_Example Gauges only have `set (value: float)`

Reply:
https://godoc.org/github.com/prometheus/client_golang/prometheus#Gauge

>  Gauge is a Metric that represents a single numerical value that can arbitrarily go up and down.
>
> A Gauge is typically used for measured values like temperatures or current memory usage, but also "counts" that can go up and down, like the number of running goroutines.